tests/compiler/test_index.py

```diff
@@ -592,3 +577,128 @@ def test_index_collision_exception(snippetcompiler) -> None:
 - index A(other_id) matches __config__::A (instantiated at {snippetcompiler.project_dir}/main.cf:17)
 """  # noqa: E501
     )
+
+
+def test_index_on_nullable(snippetcompiler) -> None:
+    """
+    Verify that indexes on nullable attributes are allowed and behave as expected.
+
+    Indexes on optional relations are not supported and are tested in test_issue_745_2689_index_on_optional.
+    """
+    snippetcompiler.setup_for_snippet(
+        """
+        entity A:
+            int? n = null
```

Comment:
this should also work if no default value is set, but then it must be set explicit in the constructor I think.

This case is not tested

Reply:
Can you elaborate? The following model works, and I believe that it should. Do you think differently?

```
entity A:
    int? n
end
index A(n)

implement A using std::none

a_null = A(n=null)
a_zero = A(n=0)


assert = true
assert = (a_null != a_zero)
assert = (a_null == A(n=null))
assert = (a_null == A[n=null])
assert = (a_zero == A(n=0))
assert = (a_zero == A[n=0])
```

Reply:
Requesting rereview to bring this to your attention

Reply:
it should

Reply:
and this one should fail

```
entity A: 
  int? n 
end 

index A(n) 

implement A using std::none 

A()
```